wasm/wasm-introduction.md

```diff
@@ -0,0 +1,157 @@
+# WASM Part I: What is WebAssembly? Reasons to take it seriously
+This article is the beginning of a series of two more posts that describes what is [WebAssembly(WASM)](https://webassembly.org/), the importance of this technology, and a simple [ClojureScript](https://clojurescript.org/) example running [WASM](https://webassembly.org/). On completion, the reader should understand the basic
+concepts related to this topic, will know the importance of [WASM](https://webassembly.org/) in the actual web, and will be capable of running
+a simple application using ClojureScript|WAT|WASM.
+## The traditional web
+The majority of the web is running using the traditional architecture, by traditional we mean
+a web browser sending requests to a server via http/s. This web browser uses HTML and CSS to render content and *delegate to JavaScript* the responsibility to manage the behavior of these components and the
+interaction between the browser and the internet via the request-response life cycle. That's it, plain
+and simple. 
+## So, this scenario is working at the moment. What is the problem?
+The problems with the traditional web are the limitations of the browser as interpreter/compiler(Depending
+on the technology implemented by the browser) and renderer at the same time. In general, the performance of
+JavaScript running resource-intensive applications like video editors, 3D games, and so on, is far away to
+compete with the native performance. 
+Remember JavaScript was designed for humans and contains rules and redundancy that must be checked by the browser to run the source code. 
+## How to solve this problem? 
+One of the many solutions for this problem could be the inclusion of a new lower-level format with a compact
+binary format to run alongside JavaScript. In this way, we could use this language to develop near-native
+performance features.
+## WebAssembly, time to raise!
+Having these ideas in mind, [WebAssembly(a.k.a WASM)](https://webassembly.org/) was launched in 2017 as a [binary instruction format](https://webassembly.github.io/spec/core/binary/index.html)
+for a **stack-based virtual machine**. We know this definition is almost the same shown at [webassembly.org](https://webassembly.org/)
+so let extend a little bit more this explanation. Let's start with **binary instruction format**, this one is simple
+because it represents the way [WASM](https://webassembly.org/) exposes instructions to the computer to be processed. On
+the other hand, we have a **stack-based virtual machine** which stores the instructions from WASM files in memory
+using a stack as the main structure(We will talk more about this later in this post).
+Now that we are understanding a little bit more the concept of WebAssembly, so let's continue with the goals and benefits this technology brings to us.
+### Main goals of WebAssembly
+Like any other technology on the web, WebAssembly has been created thinking about some specific goals in mind. We are
```

Comment:
Doesn't need both "thinking" and "in mind". Maybe "... created with some specific goals in mind."